auth/service.cc

```diff
@@ -184,8 +184,11 @@ future<> service::create_keyspace_if_missing(::service::migration_manager& mm) c
     }
 }
 
-future<> service::start(::service::migration_manager& mm) {
```

Comment:
> In the maintenance mode, scylla accepts CQL connections thus it needs the auth service.
Node is not connected to others thus schema changes are not propagated to other nodes.
If the node starts in the maintenance mode without booting previously it would create
system_auth keyspace. When the node restarts without the maintenance mode it won't
be able to join the cluster, because of schema version mismatch.

1. why would not be started in maintenance mode on first boot? Why allow it at all.
2. IIRC system distributed keyspaces are created with deterministic schema version, so they can be created on more then one node simultaneously without conflicts.

Reply:
> IIRC system distributed keyspaces are created with deterministic schema version, so they can be created on more then one node simultaneously without conflicts.

It turns out this is not true after we introduced group0_guard and started using normal timestamps taken from guard to create the mutations instead of hardcoded timestamp values like `1` or whatever it was.

Also it was true at some point for `system_distributed`, but not sure if for `system_auth`.

Reply:
> > IIRC system distributed keyspaces are created with deterministic schema version, so they can be created on more then one node simultaneously without conflicts.
> 
> It turns out this is not true after we introduced group0_guard and started using normal timestamps taken from guard to create the mutations instead of hardcoded timestamp values like `1` or whatever it was.

OK. We did well. But what about the first point?

Reply:
> 1. why would not be started in maintenance mode on first boot? Why allow it at all.

How can we distinguish if it is the first boot? We could read data from the local tables, but what if they are corrupted?

Reply:
We read data from local table a lot on boot. It would be hard to make sure non are read in maintenance mode and even if you do now it will be fragile because new reads from local tables are added constantly.

Reply:
We decided that starting scylla in the maintenance mode doesn't make sense, thus we will allow this behaviour. If it occurs then the node is broken (schema is changed only locally and the node won't be able to connect to the cluster), but since it is the first bootstrap, it can be easily rebuilt.